homeassistant/components/zeroconf/__init__.py

```diff
@@ -178,8 +182,26 @@ def service_update(zeroconf, service_type, name, state_change):
         _LOGGER.debug("Discovered new device %s %s", name, info)
 
         # If we can handle it as a HomeKit discovery, we do that here.
-        if service_type == HOMEKIT_TYPE and handle_homekit(hass, info):
-            return
+        if service_type == HOMEKIT_TYPE:
+            handle_homekit(hass, info)
+            # Continue on here as homekit_controller
```

Comment:
I am wondering here if this should be the responsibility of the `zeroconf` integration or if `homekit_controller` should register with the zeroconf ServiceBrowser directly instead.

Reply:
I think it's fine in this PR. Let's keep it.